Mathlib/Data/ZMod/Basic.lean

```diff
@@ -981,6 +987,14 @@ theorem val_eq_zero : ∀ {n : ℕ} (a : ZMod n), a.val = 0 ↔ a = 0
 theorem val_ne_zero {n : ℕ} (a : ZMod n) : a.val ≠ 0 ↔ a ≠ 0 :=
   (val_eq_zero a).not
 
+theorem val_pos_of_ne_zero {n : ℕ} {a : ZMod n} (h : a ≠ 0) : 0 < a.val :=
+  Nat.pos_of_ne_zero <| (val_ne_zero a).mpr h
+
+theorem val_eq_one : ∀ {n : ℕ} [n.AtLeastTwo] (a : ZMod n), a.val = 1 ↔ a = 1
```

Comment:
```suggestion
theorem val_eq_one : ∀ {n : ℕ} (hn : n ≠ 1) (a : ZMod n), a.val = 1 ↔ a = 1
```
Use `ZMod.val_one` and `Nat.one_mod_of_ne_one` (which you should turn into an `Iff`)

Reply:
The statement is not true for `n = 0` because `ZMod.val` then is the absolut value.

Reply:
Then
```suggestion
theorem val_eq_one : ∀ {n : ℕ} (hn : 1 < n) (a : ZMod n), a.val = 1 ↔ a = 1
```
`Nat.AtLeastTwo` should only be used for `OfNat.ofNat`